tests/libvmi/annotation.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package libvmi
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/tests/flags"
+)
+
+const (
+	HookSidecarImage                   = "example-hook-sidecar"
+	annotationKeyHookSideCars          = "hooks.kubevirt.io/hookSidecars"
+	annotationKeyBaseBoardManufacturer = "smbios.vm.kubevirt.io/baseBoardManufacturer"
+)
+
+func NewAnnotations(opts ...AnnotationsRendererOption) map[string]string {
```

Comment:
I don't think we need this new file.
We already have `WithAnnotation()` method that can be used, and in the future we should try to move towards that approach.
The only added value I see are the utility functions ` WithBaseBoardManufacturer` `WithHookSidecar`, etc..
But these are only used in the `vmi_hook_sidecar_test.go` file.
TBH I think we can define wrapper functions inside that file, without any other export.
Just my 2 cents

Reply:
@EdDev argued that in most other places we would be using builder patterns. I just reworked it to be like that. And I think this makes for quite readable code. Besides that there might be more use cases, but that is just speculative.

Reply:
But it makes sense to use `WithAnnotation`. I'll rework the code accordingly.

Reply:
Done